File: Java/Professor.java
public class Professor {

    private String nome;
    private int idade;
    private Escola escola;

    public Professor(String nome,int idade){
        this.nome = nome;
        this.idade = idade;

    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public int getIdade() {
        return idade;
    }

    public void setIdade(int idade) {
        this.idade = idade;
    }

    public Escola getEscola() {
        return escola;
    }

    public void setEscola(Escola escola) {
        this.escola = escola;
    }

    public void mostrarEscola(){
        if (escola != null)
            System.out.println("A escola "+escola.getNome()+ " contratou o professor "+ this.nome);
        else
            System.out.println(this.nome+" não trabalha nesta escola. ");
    }
}

File: Go/Professor.go
package listaRicardo

import "fmt"

type Professor struct {
	Nome   string
	Idade  int
	Escola *Escola
}

func (p *Professor) GetNome() string {
	return p.Nome
}

func (p *Professor) SetNome(nome string) {
	p.Nome = nome
}

func (p *Professor) GetIdade() int {
	return p.Idade
}

func (p *Professor) SetIdade(idade int) {
	p.Idade = idade
}

func (p *Professor) GetEscola() *Escola {
	return p.Escola
}

func (p *Professor) SetEscola(escola *Escola) {
	p.Escola = escola
}

func (p *Professor) MostrarEscola() {
	if p.Escola != nil {
		fmt.Printf("A escola %s contratou o professor %s\n", p.Escola.GetNome(), p.Nome)
	} else {
		fmt.Printf("%s não trabalha nesta escola.\n", p.Nome)
	}
}
